S/startin/build_tarballs.jl

```diff
@@ -0,0 +1,57 @@
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "startin"
+version = v"0.4.9"
+
+sources = [
+    ArchiveSource(
+        "https://github.com/evetion/startin/archive/c-interface.zip",  # PR branch
+        "d3a987e9ab20ee3504c8c79f0940ce22527e23ed8b029e8d12136831fe996c63"
+    )
+]
+
+script = raw"""
+cd $WORKSPACE/srcdir/startin-c-interface/
+if [[ "${target}" == *-darwin* ]] || [[ "${target}" == *-freebsd* ]]; then
+    # Fix linker for BSD platforms
+    sed -i "s/${rust_target}-gcc/${target}-gcc/" "${CARGO_HOME}/config"
+fi
```

Comment:
@staticfloat we've had a few issues already with this config file.  I believe we should generate it on-the-fly in `Runners.jl`, baking it in the compiler shards has lots of limitations (we don't know the OS version of BSD systems for example, which is important here).

Reply:
yeah, that's actually already done on my "rustaceans rising" branch, which never got merged because the newer Rust toolchains are broken too :(